docker/docker-frontend.sh

```diff
@@ -26,7 +26,7 @@ fi
 if [ "$BUILD_SUPERSET_FRONTEND_IN_DOCKER" = "true" ]; then
     cd /app/superset-frontend
     npm install -f --no-optional --global webpack webpack-cli
-    npm install -f --no-optional
+    npm install -f
```

Comment:
@mistercrunch I saw this flag was added previously but unsure the context. I believe there's a historical reason for it? 👀

Reply:
I did some git archeology and it looks like --no-optional goes back to https://github.com/apache/superset/pull/10766 (2020) and there's not a clear explanation. I vote going vanilla and doing a very simple `npm install` instead of those 2 lines. Webpack packages should install as its just a `devDependencies`, and they'll install normally as long as we don't specify --production, which we don't want to do here.

Reply:
tagging @eschutho as I believe she ran in an issue similar to the one you're fixing here.

Reply:
Hmm... looks like @craig-rueda added that [a looooong time ago](https://github.com/apache/superset/commit/45f4c689a2ae1e56666c36340c4b27f9f05b5903#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R76). I'm not sure if there are any consequences to its removal ¯\\\_(ツ)_/¯

Reply:
eek... duplicated effort. That's what I get for revisiting an open tab without refreshing.

Reply:
@rusackas I really think it's ok to remove this `--no-optional` as it's non-standard and in this context would only apply `BUILD_SUPERSET_FRONTEND_IN_DOCKER" = "true"` which really should align with other build context (if you're running a dev env outside docker for instance).

Reply:
This lgtm. I'll remove the pinned optional dependencies in my pr.